app/controllers/admin/communities_controller.rb

```diff
@@ -30,21 +31,40 @@ def edit_welcome_email
       :ref => "welcome_email",
       :locale => @current_user.locale
     }
+
+    sender_address = EmailService::API::Api.addresses.get_sender(community_id: @current_community.id).data[:formatted]
+    user_defined_address = EmailService::API::Api.addresses.get_user_defined(community_id: @current_community.id).data.first
+
+    render "edit_welcome_email", locals: {
+             support_email: APP_CONFIG.support_email,
+             sender_address: sender_address,
+             user_defined_address: user_defined_address,
+             post_sender_address_url: create_sender_address_admin_community_path,
+           }
+  end
+
+  def create_sender_address
+    # TODO validate email
+
+    EmailService::API::Api.addresses.create(community_id: @current_community.id, opts: {name: params[:name], email: params[:email]})
```

Comment:
opts is a quite generic and when I see it I always assume something like flags to change method behavior. How about:
```ruby
EmailService::API::Api.addresses.create(community_id: @current_community.id, address: {name: params[:name], email: params[:email]})
```